### ISSUE TITLE
AppShell Suspense error

### ISSUE THREAD

**Issue by uglyDrusLee** [mantinedev/mantine#5979]
### Dependencies check up

- [X] I have verified that I use latest version of all @mantine/* packages

### What version of @mantine/* packages do you have in package.json?

7.6.2

### What package has an issue?

@mantine/core

### What framework do you use?

Vite

### In which browsers you can reproduce the issue?

Chrome

### Describe the bug

AppShell has useResizing, which leads to rerendering, due to setState inside useIsomorphicEffect. Because of this behavior - when hydrating, react throws an error "Uncaught Error: This Suspense boundary received an update before it finished hydrating".

### Possible fix

Consider wrapping setState inside startTransition, or at least provide an option to disable useResizing. This could be particularly useful when transitionDuration is set to 0.

### Self-service

- [ ] I would be willing to implement a fix for this issue

**Comment by rtivital** [comment]
Please provide a repository with a minimal reproduction.

**Comment by uglyDrusLee** [comment]
https://github.com/uglyDrusLee/drusMantineAppShellIssue/tree/main/
here's minimal reproduction. to start, just "npm run start". server will be started on 3000 port.

**Comment by rtivital** [comment]
The issue is not related to use-resizing. If the code is chaged to 

```tsx
import { startTransition, useRef, useState } from 'react';
import { useIsomorphicEffect, useWindowEvent } from '@mantine/hooks';

interface UseResizingInput {
  transitionDuration: number | undefined;
  disabled: boolean | undefined;
}

export function useResizing({ transitionDuration, disabled }: UseResizingInput) {
  const [resizing, setResizing] = useState(true);
  const resizingTimeout = useRef<number>();
  const disabledTimeout = useRef<number>();

  useWindowEvent('resize', () => {
    setResizing(true);
    clearTimeout(resizingTimeout.current);
    resizingTimeout.current = window.setTimeout(() => setResizing(false), 200);
  });

  useIsomorphicEffect(() => {
    startTransition(() => {
      setResizing(true);
      clearTimeout(disabledTimeout.current);
      disabledTimeout.current = window.setTimeout(
        () => setResizing(false),
        transitionDuration || 0
      );
    });
  }, [disabled, transitionDuration]);

  return resizing;
}
```

Or resizing removed entirely, it does not fix the error message. 

You are welcome to research the issue and propose the fix.

**Comment by uglyDrusLee** [comment]
your solution works fine for me, error is gone. maybe you faced vite's cache? to get rid of it you can just delete ".vite" folder inside node_modules, restart "npm run start". dont know how exactly you tested with my repo. i just manually edit useResizing inside node_modules and error is gone.

**Comment by rtivital** [comment]
All right, then I'll just add this solution in next patch

**Comment by rtivital** [comment]
Fixed in [7.1.1](https://github.com/mantinedev/mantine/releases/tag/7.7.1)

**Comment by RabeeAbuBaker** [comment]
Getting this error again on `v7.13.0`

```
Uncaught Error: This Suspense boundary received an update before it finished hydrating. This caused the boundary to switch to client rendering. The usual way to fix this is to wrap the original update in startTransition.
```

Using `Suspense` outside of the `AppShell` doesn't trigger the error

![image](https://github.com/user-attachments/assets/4a891dc7-c640-41bb-b60b-a178257526ed)
